### ISSUE TITLE
Search button functionality across the site

### ISSUE THREAD

**Issue by cbenea** [dotnet/docfx#2679]
**DocFX Version Used**: 2.21.1.1

**Template used**: custom template

**Steps to Reproduce**:

- script in docfx.js?

**Expected Behavior**:
Search button should return and highlight all matching words from the entire site.

**Actual Behavior**:
Search button doesn't return any result, no other message is displayed

**Comment by qinezh** [comment]
Is there a file named `index.json` generated by DocFX? I got a 404 while accessing your webiste with this URL:http://developer.temenos.com/index.json.   

For other example, like documentation site for DocFX, the `index.json` could be accessed via http://dotnet.github.io/docfx/index.json.

**Comment by Still34** [comment]
I'd like to chime in on this issue as well - our docfx does generate `index.json`; however, the search always returns nothing. Our [documentation preview](https://docs.stillu.cc). This was working several revisions before.

**Comment by cbenea** [comment]
Hi! 
I searched for index.json but indeed it seems I don't have it. 
What should I do? 
I really appreciate your help to solve this. Thank you!

**Comment by Still34** [comment]
No, but no changes after enabling it.

**Comment by cbenea** [comment]
I enabled it now and it works for me. Thank you very much!

**Comment by superyyrrzz** [comment]
@Still34 Have you try build your documents with default template? I find there is an error in JS when rendering the page, whose source is from your custom template. The error may affect other JS, which search feature relies on.

> Uncaught SyntaxError: Identifier 'baseUrl' has already been declared
>     at styleswitcher.js:1

**Comment by Still34** [comment]
@superyyrrzz Yes, the search stopped working before the recent theme switcher implementation.

**Comment by superyyrrzz** [comment]
@Still34 Could you try fixing the error in the switcher then see if the search will work?

**Comment by Still34** [comment]
@superyyrrzz Found the reason. DocFX ships with an older version of jQuery with `docfx.vendor.js`, and our theme references a newer version so DocFX got confused.

**Comment by herohua** [comment]
@Still34 Looks like the root cause of the issue has been identified. Do you need more help from DocFX side?

@superyyrrzz as FYI
